Storage.md

```diff
@@ -0,0 +1,230 @@
+This guide will describe how FullScreenShenanigans projects keep track of game information.
+
+### Table of Contents
+1. [ItemsHoldr](#itemsholdr)
+    1. [Local Storage](#local-storage)
+    2. [Items](#items)
+    3. [Updating](#updating)
+    4. [Clearing and Defaults](#clearing-and-defaults)
+    5. [Elements](#elements)
+2. [StateHoldr](#stateholdr)
+
+# ItemsHoldr
+
+ItemsHoldr is a versatile container to store and manipulate values in localStorage, and optionally keep an updated HTML container showing these values.
+
+## Local Storage
+
+`localStorage` is a browser property where information is saved without expiring.
+ItemsHoldr keeps track of items and their values while the game is running.
+
+Other sources (e.g. websites) also use `localStorage`, so ItemsHoldr has a prefix property to differentiate which information belongs to which.
+
+```typescript
+let ItemsHolder: ItemsHoldr = new ItemsHoldr({
+    prefix: "FullScreenShenanigans"
+});
+```
+
+## Items
+
+To add an item to the collection, use `addItem` and to remove an item, use `removeItem`.
+
+```typescript
+ItemsHolder.addItem("color", { value: "red" });
+ItemsHolder.removeItem("color");
+```
+
+`setItem` is used to set the value of an item.
+
+```typescript
+ItemsHolder.setItem("color", "purple");
+```
+
+To manually update a changed item's value in localStorage, use `saveItem`.
+To save all the items' values in localStorage, use `saveAll`.
+
+```typescript
+ItemsHolder.saveItem("color");
+ItemsHolder.saveAll();
+```
+
+`increase` and `decrease` add and subtract respectively from the item's value.
+`toggle` flips the boolean value of the item.
+
+```typescript
+ItemsHolder.increase("weight", 14);
+ItemsHolder.decrease("weight", 10);
+ItemsHolder.toggle("started");
+```
+
+`checkExistence` can be used to check if there is an item under the specified key and if not, add it to the collection.
+
+```typescript
+ItemsHolder.checkExistence("color");
+```
+
+By default if an item does not exist when `setItem`, `getItem`, `increase`, or `toggle` are called, the item is then added to the collection in the process of the respective function.
+Define `allowNewItems` to explicitly enable or disable this feature.
+
+```typescript
+let ItemsHolder: ItemsHoldr = new ItemsHoldr({ allowNewItems: false });
+```
+
+## Updating
+
+When an item's value changes, the module checks its triggers, modularity, and element status.
+
+Triggers are functions that get run when an item's value equals the specified value.
+
+```typescript
+ItemsHolder.addItem("color", {
+    triggers: {
+        red: () => { this.value = "black"; }
+    }
+});
+ItemsHolder.setItem("color", "red");
+```
+
+Modularity restricts an item's value to a range from 0 to the specified max and runs the `onModular` function as many times until the value hits the modular range.
+
+```typescript
+ItemsHolder.addItem("time", {
+    modularity: 12,
+    num: 0,
+    onModular: () => { this.num += 1; }
```

Comment:
That's a good idea, but doesn't work well as implemented now because `num` doesn't relate to time.